runtime/src/main/kotlin/kotlin/native/concurrent/Freezing.kt

```diff
@@ -42,12 +46,37 @@ public fun <T> T.freeze(): T {
 public val Any?.isFrozen
     get() = isFrozenInternal(this)
 
+
+/**
+ * Transforms an object to the frozen form by creating frozen version of the object, by
+ * deep copy of mutable objects and reusing frozen references. Note, that consequent calls
+ * of this function may return different instances. To make a clone class must be eitehr data class or
+ * explicitly marked by [kotlin.ShareByValue] annotation.
+ *
+ * @return the frozen object form
+ * @see ensureNeverFrozen
+ * @see ensureNeverCloned
+ * @see kotlin.ShareByValue
+ */
+public inline fun <reified T> T.toFrozen(): T =
```

Comment:
> ```text
> To make a clone class must be either data class or explicitly marked by [kotlin.ShareByValue] annotation.
> ```

What if this was generated _only_ for the types that support it?
Either directly on the types (like the `copy()` function in data classes), or by having the types implement a special internal marker interface? The second option would actually be ideal, since it would allow targeting any cloneable type!